File: Models/Role.cs
using System.ComponentModel.DataAnnotations;
using MediCare.Enums;

namespace MediCare.Models
{
	public class Role : Entity
	{
		[Required]
		public RoleType RoleType { get; set; }

		public ICollection<Permission> Permissions { get; set; } = new List<Permission>();
	}
}

File: MediCareWebSpa/src/app/DTOs/models/role.ts
import { RoleType } from "../../enums/role-type";
import { RolePermission } from "./role-permission";

export class Role {
  id: number;
  roleType: RoleType;
  rolePermissions: RolePermission[] = [];

  constructor(
    id: number,
    roleType: RoleType
  ) {
    this.id = id;
    this.roleType = roleType;
  }
}
